### DISCUSSION TITLE
Patch a collection & update ids

### DISCUSSION THREAD

**Discussion by ItsRobbAllen** [ravendb/ravendb#15453] [Q&A]
The short story is I have 2 objects that are misnamed & I need to rename both of them & change the collections the existing data is in. To make things more convoluted, one object needs to be renamed to the other one.

It's a growing pain of a new system.  We currently have ActiveRoyaltyPlan and RoyaltyPlan. What we need is `RoyaltyPlan` and `RoyaltyPlanTemplate`. Originally the object we are using called RoyaltyPlan was the actualized plan, but it morphed into being just a template. The `ActiveRoyaltyPlan` has a `Plan` property which contains a full copy of the `RoyaltyPlan`.

What I want to do is to patch all the existing `RoyaltyPlan` objects into the `RoyaltyPlanTemplates` collection which is easy enough to do with 
```
from RoyaltyPlans as rp
                  update {
                      del(id(rp));
                      this["@metadata"]["@collection"] = "RoyaltyPlanTemplates";
                      put(id(rp), this);
                  }
```
But that recreates the documents with the exact same ID, which I don't want. I'm perfectly ok with brand new ID's like `RoyaltyPlanTemplates/1-A` as the system isn't in use enough right now to need to worry about issues.

How do I patch a document, move it to a new collection, have have a new ID generated for it?

**Comment by ayende** [comment]
` put('RoyaltyPlanTemplates/', this);`

This will generate new ids for you.
